### DISCUSSION TITLE
IndexSubset for y

### DISCUSSION THREAD

**Discussion by TeoSkondras** [alan-turing-institute/sktime#4994] [Q&A]
through IndeSubset we can adjust X on y.index. How can we do the reverse; change the y to have index values that are inside X.index? I would like to add it to a TransformedTargetForecaster.

What I do is that I create a pipe_y and then feed it a pipe_x exactly like the example with: https://github.com/sktime/sktime/blob/main/examples/03b_forecasting_transformers_pipelines_tuning.ipynb

Is there anything I can do to keep only the y indexes that are in X.index (their intersection/ inner join)?

**Comment by fkiraly** [comment]
To apply transformers to `y`, use a `ForecastingPipeline`. This plugs the forecasting `y` into the transformer `X`, and the forecasting `X` in the transformer `y` - i.e., the opposite way round from the `TransformedTargetForecaster`.

Let us know if this helped!

**Reply by fkiraly** [reply]
(for a shorthand, the `ForecastingPipeline` can be built via the `*` dunder, and the `TransformedTargetForecaster` via the `**` dunder)

**Reply by TeoSkondras** [reply]
So you are suggesting that we first start by transforming the y through the ForecastingPipeline and then feed it to the TransformedTargetPipeline? (We currently do the opposite as the example suggests)
